### PR TITLE
[Not ready] storage circuits selfreference fix

### PR DESCRIPTION
**Pull request by Shdorsh**
Back then, the self-reference output pin didn't work: it didn't reference the circuit itself from the start and had to be set up to do so. Now, each time a storage circuit is spawned, it automatically references itself, as it was supposed to.

[Changelogs]: # Fixed selfreference in most beaker-like circuits

:cl: Shdorsh
fix: selfreference now is initially set to itself for most circuits, as it was intended to (see: name of the pin)
/:cl:

[why]: # First, because that's what it was supposed to do I believe. Secondly, it makes handling reagent circuitry easier.

Edit: I see this thing isn't as good as it could be so I set it to not ready and will work on raising it to something a little bit more respectable.

### CODE REVIEW

File: code/modules/integrated_electronics/subtypes/reagents.dm [hunk 1]
```diff
@@ -43,6 +43,10 @@
 	var/smoke_radius = 5
 	var/notified = FALSE
 
+/obj/item/integrated_circuit/reagent/smoke/New()
+	..()
```
Comment (Cyberboss) [inline code/modules/integrated_electronics/subtypes/reagents.dm]: Initialize and . = ..()

File: code/modules/integrated_electronics/subtypes/reagents.dm [hunk 2]
```diff
@@ -133,6 +137,10 @@
 	var/transfer_amount = 10
 	var/busy = FALSE
 
+/obj/item/integrated_circuit/reagent/injector/New()
+	..()
```
Comment (Cyberboss) [inline code/modules/integrated_electronics/subtypes/reagents.dm]: Same

File: code/modules/integrated_electronics/subtypes/reagents.dm [hunk 3]
```diff
@@ -325,7 +333,9 @@
 	activators = list("push ref" = IC_PINTYPE_PULSE_OUT)
 	spawn_flags = IC_SPAWN_DEFAULT|IC_SPAWN_RESEARCH
 
-
+/obj/item/integrated_circuit/reagent/storage/New()
+	..()
```
Comment (Cyberboss) [inline code/modules/integrated_electronics/subtypes/reagents.dm]: Same

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/integrated_electronics/subtypes/reagents.dm [hunk 4]
```diff
@@ -16,59 +16,6 @@
 	set_pin_data(IC_OUTPUT, 1, reagents.total_volume)
 	push_data()
 
-/obj/item/integrated_circuit/reagent/smoke
```
Comment (ShizCalev) [inline code/modules/integrated_electronics/subtypes/reagents.dm]: why did you delete this?

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by PKPenguin321** [comment]
Sounds like it's supposed to make pulsing the "push ref" pulse in unnecessary, as it will automatically self reference on creation. That said this PR is completely nonfunctional until he fixes his merge conflict business

**Comment by Shdorsh** [comment]
The merge conflict was long gone, as I removed the smoke circuit from it, which was the part that dealed the trouble. I'll add some more stuff later on to make this work even better.

File: code/modules/integrated_electronics/subtypes/reagents.dm (hunk 4 above)
Comment (Shdorsh) [reply code/modules/integrated_electronics/subtypes/reagents.dm]: The circuit nerf from a few weeks ago removed smoke circuits entirely, while this PR still had the code of the smoke circuit in it, since this was written even before the nerf. As such, this would have overwritten the nerf and reintroduced smoke circuits, hence the merge conflict.

**Comment by Shdorsh** [comment]
I'll just close this down entirely and make something new, I have a different way to do this in mind, but it will require entirely different changes, so it would be best to change from scrap.
